azure-mgmt-resources/src/test/java/com/microsoft/azure/management/resources/core/ResourceGroupTaggingInterceptor.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.core;
+
+import com.microsoft.azure.management.resources.implementation.ResourceGroupInner;
+import com.microsoft.azure.serializer.AzureJacksonAdapter;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okio.Buffer;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An interceptor for tagging resource groups created in tests.
+ */
+public class ResourceGroupTaggingInterceptor implements Interceptor {
+    private static final String LOGGING_CONTEXT = "com.microsoft.azure.management.resources.ResourceGroups createOrUpdate";
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        if ("PUT".equals(chain.request().method()) && chain.request().url().uri().toString().contains("/resourcegroups/") &&
+                LOGGING_CONTEXT.equals(chain.request().header("x-ms-logging-context"))) {
+            String body = bodyToString(chain.request());
+            AzureJacksonAdapter adapter = new AzureJacksonAdapter();
+            ResourceGroupInner rg = adapter.deserialize(body, ResourceGroupInner.class);
```

Comment:
Should we worry about failure in adapter.deserialize method? 

if adapter.deserialize throws then we are good, we get exact reason why its failed. 

If it return null on deserialization failure then we might want to throw a runtime exception indicating deserialize failed instead of java throwing NPE in line 36.

We can test this by creating AzureJacksonAdapter instance in a test, and then try to deserialize an invalid string which is not conforming to resource group schema and see it throws vs return null.